inc/adc_interface_client.hpp

```diff
@@ -31,18 +31,26 @@ class AdcInterfaceClient : public ClientAbstract {
     ClientEntry<float> adc_voltage_;
     ClientEntry<uint16_t> raw_value_;
 
-    void ReadMsg(uint8_t* rx_data, uint8_t rx_length) {
-        static const uint8_t kEntryLength              = kSubRawValue + 1;
-        ClientEntryAbstract* entry_array[kEntryLength] = {
-            &adc_voltage_,  // 0
-            &raw_value_     // 1
-        };
-        ParseMsg(rx_data, rx_length, entry_array, kEntryLength);
+    uint16_t GetNumberOfClientEntries(){
+      return kSubRawValue + 1;
     }
 
+   void GetClientEntryList(ClientEntryAbstract ** client_entries){
+      uint16_t num_entries = GetNumberOfClientEntries();
+
+      ClientEntryAbstract* entry_array[num_entries] = {
+          &adc_voltage_,  // 0
+          &raw_value_     // 1
+      };
+
+      for(uint16_t entry = 0; entry < num_entries; entry++){
```

Comment:
I'm not actually sure that this would be any better, fairly low priority thing, but I'll just throw it out there as it may be a bit easier than having lots of for loops: Would using something like memcpy to copy over the list here make sense? May make this part a single-liner with a standard function. Again, just a random thing to consider